### ISSUE TITLE
Please update embedded javac version

### ISSUE THREAD

**Issue by don-vip** [google/error-prone#451]
We're using error-prone 2.0.11 with ant and face the following error:

```
Apache Ant(TM) version 1.9.7 compiled on April 9 2016
Trying the default build file: build.xml
Buildfile: C:\SVN\josm\plugins\dxfimport\build.xml
Detected Java version: 1.8 in: C:\Program Files\Java\jdk1.8.0_102\jre
Detected OS: Windows 10
...
[javac] Compiling 5 source files to C:\SVN\josm\plugins\dxfimport\build
[javac] Compilation arguments:
[javac] '-d'
[javac] 'C:\SVN\josm\plugins\dxfimport\build'
[javac] '-classpath'
[javac] 'C:\SVN\josm\plugins\dxfimport\build;C:\SVN\josm\core\dist\josm-custom.jar;C:\SVN\josm\plugins\dxfimport\lib\batik-dom.jar;C:\SVN\josm\plugins\dxfimport\lib\batik-ext.jar;C:\SVN\josm\plugins\dxfimport\lib\batik-parser.jar;C:\SVN\josm\plugins\dxfimport\lib\batik-svgpp.jar;C:\SVN\josm\plugins\dxfimport\lib\batik-transcoder.jar;C:\SVN\josm\plugins\dxfimport\lib\batik-util.jar;C:\SVN\josm\plugins\dxfimport\lib\batik-xml.jar;C:\SVN\josm\plugins\dxfimport\lib\kabeja-0.4.jar;C:\SVN\josm\plugins\dxfimport\lib\kabeja-svg-0.4.jar;C:\SVN\josm\plugins\dxfimport\lib\kabeja-xslt.jar'
[javac] '-sourcepath'
[javac] 'C:\SVN\josm\plugins\dxfimport\src'
[javac] '-target'
[javac] '1.8'
[javac] '-encoding'
[javac] 'UTF-8'
[javac] '-g'
[javac] '-Xlint:deprecation'
[javac] '-Xlint:unchecked'
[javac] '-Xep:ReferenceEquality:OFF'
[javac] '-Xmaxwarns'
[javac] '1000'
[javac] '-source'
[javac] '1.8'
[javac]
[javac] The ' characters around the executable and arguments are not part of the command.
[javac] Files to be compiled:
[javac]     C:\SVN\josm\plugins\dxfimport\src\org\openstreetmap\josm\plugins\dxfimport\DxfImportPlugin.java
[javac]     C:\SVN\josm\plugins\dxfimport\src\org\openstreetmap\josm\plugins\dxfimport\DxfImportTask.java
[javac]     C:\SVN\josm\plugins\dxfimport\src\org\openstreetmap\josm\plugins\dxfimport\DxfImporter.java
[javac]     C:\SVN\josm\plugins\dxfimport\src\org\openstreetmap\josm\plugins\dxfimport\ImportDialog.java
[javac]     C:\SVN\josm\plugins\dxfimport\src\org\openstreetmap\josm\plugins\dxfimport\Settings.java
[javac] An exception has occurred in the compiler (1.9.0-internal). Please file a bug at the Java Bug Database (http://bugreport.java.com/bugreport/) after checking the database for duplicates. Include your program and the following diagnostic in your report.  Thank you.
[javac] java.lang.NullPointerException
[javac]     at com.sun.tools.javac.file.JavacFileManager.listContainer(JavacFileManager.java:399)
[javac]     at com.sun.tools.javac.file.JavacFileManager.list(JavacFileManager.java:676)
[javac]     at com.sun.tools.javac.code.ClassFinder.scanUserPaths(ClassFinder.java:564)
[javac]     at com.sun.tools.javac.code.ClassFinder.fillIn(ClassFinder.java:504)
[javac]     at com.sun.tools.javac.code.ClassFinder.complete(ClassFinder.java:287)
[javac]     at com.sun.tools.javac.code.ClassFinder.access$000(ClassFinder.java:72)
[javac]     at com.sun.tools.javac.code.ClassFinder$1.complete(ClassFinder.java:159)
[javac]     at com.sun.tools.javac.code.Symbol.complete(Symbol.java:579)
[javac]     at com.sun.tools.javac.comp.Enter.visitTopLevel(Enter.java:299)
[javac]     at com.sun.tools.javac.tree.JCTree$JCCompilationUnit.accept(JCTree.java:509)
[javac]     at com.sun.tools.javac.comp.Enter.classEnter(Enter.java:255)
[javac]     at com.sun.tools.javac.comp.Enter.classEnter(Enter.java:270)
[javac]     at com.sun.tools.javac.comp.Enter.complete(Enter.java:483)
[javac]     at com.sun.tools.javac.comp.Enter.main(Enter.java:467)
[javac]     at com.sun.tools.javac.main.JavaCompiler.enterTrees(JavaCompiler.java:952)
[javac]     at com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:833)
[javac]     at com.sun.tools.javac.main.Main.compile(Main.java:253)
[javac]     at com.google.errorprone.BaseErrorProneCompiler.run(BaseErrorProneCompiler.java:190)
[javac]     at com.google.errorprone.BaseErrorProneCompiler.run(BaseErrorProneCompiler.java:101)
[javac]     at com.google.errorprone.ErrorProneCompiler.run(ErrorProneCompiler.java:131)
[javac]     at com.google.errorprone.ErrorProneCompiler.compile(ErrorProneCompiler.java:73)
[javac]     at com.google.errorprone.ErrorProneAntCompilerAdapter$AntRunner.apply(ErrorProneAntCompilerAdapter.java:39)
[javac]     at com.google.errorprone.ErrorProneAntCompilerAdapter$AntRunner.apply(ErrorProneAntCompilerAdapter.java:36)
[javac]     at com.google.errorprone.ErrorProneAntCompilerAdapter.execute(ErrorProneAntCompilerAdapter.java:72)
[javac]     at org.apache.tools.ant.taskdefs.Javac.compile(Javac.java:1344)
[javac]     at org.apache.tools.ant.taskdefs.Javac.execute(Javac.java:1070)
[javac]     at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:293)
[javac]     at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
[javac]     at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[javac]     at java.lang.reflect.Method.invoke(Method.java:498)
[javac]     at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
[javac]     at org.apache.tools.ant.Task.perform(Task.java:348)
[javac]     at org.apache.tools.ant.Target.execute(Target.java:435)
[javac]     at org.apache.tools.ant.Target.performTasks(Target.java:456)
[javac]     at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1405)
[javac]     at org.apache.tools.ant.Project.executeTarget(Project.java:1376)
[javac]     at org.apache.tools.ant.helper.DefaultExecutor.executeTargets(DefaultExecutor.java:41)
[javac]     at org.apache.tools.ant.Project.executeTargets(Project.java:1260)
[javac]     at org.apache.tools.ant.Main.runBuild(Main.java:854)
[javac]     at org.apache.tools.ant.Main.startAnt(Main.java:236)
[javac]     at org.apache.tools.ant.launch.Launcher.run(Launcher.java:285)
[javac]     at org.apache.tools.ant.launch.Launcher.main(Launcher.java:112)
```

It appears to be an old Java 9 bug which has been fixed in b108 and released in March 2016: https://bugs.openjdk.java.net/browse/JDK-8150475

Can you please update the javac version so this fix is included? Thanks.

**Comment by eaftan** [comment]
We can't update to javac9 at head because it now requires to run on JDK9.  We're working on a solution for this, but it won't be ready until the end of the year at the earliest.  Can you work around this?

**Comment by don-vip** [comment]
OK. Easiest way is to backport the fix, it is trivial: https://github.com/google/error-prone-javac/pull/2

**Comment by Stephan202** [comment]
(This comment is mostly for documentation purposes, in case others hit this issue as well; I'm sure the Error Prone team is aware of the need to upgrade the embedded javac.)

---

I hit the following issue when running with Error Prone, but not with stock JDK 1.8.0_112, nor with 9-ea+149:
```
java.lang.IllegalAccessError: tried to access class com.example.foo.AbstractDummy from class com.example.bar.DummyTest
	at com.example.bar.DummyTest.lambda$test$0(DummyTest.java:12)
	at com.example.bar.DummyTest.test(DummyTest.java:12)`
```

This appears to be issue [JDK-8138667](https://bugs.openjdk.java.net/browse/JDK-8138667), and happens when using a method reference to a public method in a package-private class in some other package, made available through a public subclass. E.g.

1. `com.example.foo.AbstractDummy`:
    ```java
    package com.example.foo;

    abstract class AbstractDummy {
        public final int getInt() {
            return 0;
        }   
    }   
    ``` 
2. `com.example.foo.Dummy`:
    ```java
    package com.example.foo;

    public final class Dummy extends AbstractDummy {}
    ``` 
3. `com.example.bar.DummyTest`:
    ```java
    package com.example.bar;

    import static org.testng.Assert.assertEquals;

    import com.example.foo.Dummy;
    import java.util.stream.Stream;
    import org.testng.annotations.Test;

    public final class DummyTest {
        @Test
        public void test() {
            assertEquals(Stream.of(new Dummy()).mapToInt(Dummy::getInt).sum(), 0);
        }
    }
    ``` 

When you hit this issue, either:
1. Make the superclass (`AbstractDummy`) public; or
2. Replace the method reference (`Dummy::getInt`) with a lambda expression (`d -> d.getInt()`).

**Comment by Stephan202** [comment]
Or should I file a new ticket, asking for [JDK-8138667](https://bugs.openjdk.java.net/browse/JDK-8138667) to be backported?

**Comment by cushon** [comment]
JDK-8150475 is fixed in Error Prone 2.0.16.

**Comment by cushon** [comment]
JDK-8150475 is fixed in Error Prone 2.0.16.
